test/uq_parameters.jl

```diff
@@ -0,0 +1,224 @@
+            [Parameterized(Gamma(2.0, 1.0)), Parameterized(Gamma(2.0, 1.0)),
+             Parameterized(Gamma(2.0, 1.0))],
+            [[bounded_above(9.0)], [bounded_above(9.0)], [bounded_above(9.0)]],
+            ["uq_param_6_(1)", "uq_param_6_(2)", "uq_param_6_(3)"]),
+
+        "uq_param_7" => ParameterDistribution(
+            Parameterized(MvNormal(3, 2.0)),
+            [no_constraint(), no_constraint(), no_constraint()],
+            "uq_param_7")
+    )
+
+    # Get all `ParameterDistribution`s. We also add dummy (key, value) pairs
+    # to check if that information gets added correctly when saving the
+    # parameters back to file and re-loading them
+    descr = " will be learned using CES"
+    for param_name in keys(param_dict)
+        param_dict[param_name]["description"] = param_name * descr
+        pd = CP.get_parameter_distribution(param_dict, param_name)
+        target_pd = target_map[param_name]
+
+        # Check names
+        @test get_name(pd) == get_name(target_pd)
+        # Check distributions
+        @test get_distribution(pd) == get_distribution(target_pd)
+        # Check constraints
+        constraints = get_all_constraints(pd)
+        target_constraints = get_all_constraints(target_pd)
+        @test constraints == target_constraints
+
+    end
+
+    # We can also get a `ParameterDistribution` representing
+    # multiple parameters
+    param_names = ["uq_param_2", "uq_param_4", "uq_param_5"]
+    pd = CP.get_parameter_distribution(param_dict, param_names)
+
+    # Save the parameter dictionary and re-load it. 
+    logfile_path = joinpath(@__DIR__,"log_file_test_uq.toml")
+    CP.write_log_file(param_dict, logfile_path)
+
+    # Read in log file as new parameter file and rerun test.
+    param_dict_from_log = CP.parse_toml_file(logfile_path)
+    rm(logfile_path) # clean up
+    for param_name in keys(param_dict_from_log)
+        pd = CP.get_parameter_distribution(param_dict_from_log, param_name)
+        @test get_distribution(pd) == get_distribution(target_map[param_name])
+        @test param_dict_from_log[param_name]["description"] == param_name*descr
+    end
+
+end
+
+# This test set creates a directory "test_output" in "CLIMAParameters.jl/test"
+# where the new parameteter files resulting from updating an ensemble Kalman
+# ensemble are saved
+@testset "Save parameter ensemble" begin
+
+    # Combine a parameter struct from the parameters defined in
+    # "uq_test_parameters.toml" (the override file) and those defined in
+    # "test_parameters.toml" (the default file)
+    param_path = joinpath(@__DIR__, "test_parameters.toml")
+    uq_param_path = joinpath(@__DIR__,"uq_test_parameters.toml")
+    param_set = CP.create_parameter_struct(
+        uq_param_path,
+        param_path,
+        dict_type="name"
+    )
+
+    # Extract the UQ parameters from the joint set of the parameters from
+    # param_path and those from uq_param_path
+    uq_param_names = CP.get_UQ_parameters(param_set)
```

Comment:
I'm not sure this is the right use-case
```julia
param_set = CP.create_parameter_struct(
        uq_param_path,
        param_path,
        dict_type="name"
    )
uq_param_names = CP.get_UQ_parameters(param_set)
```
I think we should have 
1. a single file with both uq and other parameters in it
2. a toml parsing function in `file_parsing_uq.jl` to load up the parameters from the toml file 
Currently this code uses the `create_param_struct` from CLIMAParameters which does additional loading and overriding of defaults.

Reply:
If you are only using this in the test case - that is 'ok' but could you perhaps make it more explicit / provide the `file_parsing_uq.jl` reader for the toml file?

Reply:
I added a parse function `read_parameters` to `file_parsing_uq.jl`. `read_parameters` only takes a single argument, the path to the parameter file (a .toml file) that is to be read. No override / default combinations and distinctions; the assumption is that all parameters (parameters to be calibrated and other ones) live in a single file.

Reply:
Great! I'll take a look